examples/chip-tool-darwin/commands/common/CHIPCommandBridge.h

```diff
@@ -0,0 +1,94 @@
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+#pragma once
+#import <CHIP/CHIPDeviceController.h>
+#include <commands/common/Command.h>
+#include <commands/common/CredentialIssuerCommands.h>
+
+#pragma once
+
+class CHIPCommandBridge : public Command
+{
+public:
+    CHIPCommandBridge(const char * commandName) : Command(commandName)
+    {
+#if CHIP_CONFIG_TRANSPORT_TRACE_ENABLED
+        AddArgument("trace_file", &mTraceFile);
+        AddArgument("trace_log", 0, 1, &mTraceLog);
+#endif // CHIP_CONFIG_TRANSPORT_TRACE_ENABLED
+    }
+
+    CHIPCommandBridge(const char * commandName, CredentialIssuerCommands * credIssuerCmds) : CHIPCommandBridge(commandName) {}
+
+    /////////// Command Interface /////////
+    CHIP_ERROR Run() override;
+
+    void SetCommandExitStatus(CHIP_ERROR status)
+    {
+        mCommandExitStatus = status;
+        ShutdownCommissioner();
+        StopWaiting();
+    }
+
+protected:
+    // Will be called in a setting in which it's safe to touch the CHIP
+    // stack. The rules for Run() are as follows:
+    //
+    // 1) If error is returned, Run() must not call SetCommandExitStatus.
+    // 2) If success is returned Run() must either have called
+    //    SetCommandExitStatus() or scheduled async work that will do that.
+    virtual CHIP_ERROR RunCommand() = 0;
+
+    // Get the wait duration, in seconds, before the command times out.
+    virtual chip::System::Clock::Timeout GetWaitDuration() const = 0;
+
+    // Shut down the command, in case any work needs to be done after the event
+    // loop has been stopped.
+    virtual void Shutdown() {}
+
+    void SetIdentity(const char * name);
+
+    // This method returns the commissioner instance to be used for running the command.
+    CHIPDeviceController * CurrentCommissioner();
+
+private:
+    CHIP_ERROR InitializeCommissioner(std::string key, chip::FabricId fabricId);
+    CHIP_ERROR ShutdownCommissioner();
+    uint16_t CurrentCommissionerIndex();
+
+    CHIP_ERROR mCommandExitStatus = CHIP_ERROR_INTERNAL;
+
+    CHIP_ERROR StartWaiting(chip::System::Clock::Timeout seconds);
+    void StopWaiting();
+    CHIPDeviceController * mController;
+
+#if CONFIG_USE_SEPARATE_EVENTLOOP
+    std::condition_variable cvWaitingForResponse;
+    std::mutex cvWaitingForResponseMutex;
+    bool mWaitingForResponse{ true };
+#endif // CONFIG_USE_SEPARATE_EVENTLOOP
```

Comment:
humm. I suspect that for `chip-tool-darwin` we instead want to do something with `[[NSRunLoop currentRunLoop] run];` and use the default run loop behaviour.

But we can probably move forward with that at the moment !

Reply:
Ok. Sounds good.

Reply:
I created an issue: https://github.com/project-chip/connectedhomeip/issues/14734